packages/hyperion-autologging/src/ALInteractableDOMElement.ts

```diff
@@ -182,16 +182,83 @@ const extractInnerText = (element: HTMLElement | null): string | null => {
   return null;
 };
 
+export type ALElementNameResult = Readonly<{
+  text: string,
+  source: 'innerText' | 'aria-label' | 'aria-labelledby'| 'aria-description' | 'aria-describedby';
+}>;
+
 export function getElementName(
   element: HTMLElement,
   // Whether to try to resolve if element name is a result of Fbt translation
-): string | null {
+): ALElementNameResult | null {
   let nextElement: HTMLElement | null = element;
   while (nextElement && nextElement.nodeType === Node.ELEMENT_NODE) {
+
+    const labelledBy = nextElement.getAttribute('aria-labelledby');
+    if (labelledBy != null) {
+      const fullLabel = labelledBy.split(' ').map(labelledBy => {
+        const labelElement = document.getElementById(labelledBy);
+        if (labelElement != null) {
+          const labelText = extractInnerText(labelElement);
+          if (labelText != null) {
+            return {
+              text: labelText,
+            };
+          }
+        }
+        return null;
+      });
+      const fullLabelText = fullLabel.map(label => label?.text).join(' ');
+      if (fullLabelText != null && fullLabelText !== '') {
+        return {
+          text: fullLabelText,
+          source: 'aria-labelledby',
+        };
+      }
+    }
+
+    const label = nextElement.getAttribute('aria-label');
+    if (label != null && label !== '') {
+        return {text: label, source: 'aria-label'};
+    }
+
+    const description = nextElement.getAttribute('aria-description');
+    if (description != null && description !== '') {
+      return {
+        text: description,
+        source: 'aria-description',
+      };
+    }
+
+    const describedBy = nextElement.getAttribute('aria-describedby');
+    if (describedBy != null) {
+      const fullDesc = describedBy.split(' ').map(describedBy => {
+        const descElement = document.getElementById(describedBy);
+        if (descElement != null) {
+          const descText = extractInnerText(descElement);
+          if (descText != null) {
+            return {
+              text: descText,
+            };
+          }
+        }
+        return null;
+      });
+      const fullDescText = fullDesc.map(desc => desc?.text).join(' ');
```

Comment:
same comment for here. In fact, you can turn this into a function that receives the whole string, with spaces, and returns the final string.